extensions/api/control-rest/src/main/java/org/eclipse/dataspaceconnector/api/rest/ApiController.java

```diff
@@ -0,0 +1,137 @@
+package org.eclipse.dataspaceconnector.api.rest;
+
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import org.eclipse.dataspaceconnector.common.collection.CollectionUtil;
+import org.eclipse.dataspaceconnector.metadata.catalog.CatalogService;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.response.ResponseStatus;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.TransferProcessStates;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.UUID;
+
+@Consumes({MediaType.APPLICATION_JSON})
+@Produces({MediaType.APPLICATION_JSON})
+@Path("/")
+public class ApiController {
```

Comment:
Is this controller thought to be the central REST interface for controlling the connector?

Reply:
Yep. I suppose in many cases we need to interact with a connector, so I figured it would be a usable starting point.

Reply:
hm, most of the endpoints are tightly coupled to what's happening in the core. Maybe it should be part of it?
At least narrow the scope: is it about providing adminstrative endpoints? Is is just for demo/test/... reasons? Can we split it up by functional groups (monitoring, data transfer, data catalog) ...

Reply:
we can discuss/do all that, but this pull request was intended to move out the hackathon code away from the `samples/gaiax-hackathon-1` folder. Let's open another issue for this?

Reply:
https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/79

Reply:
Would it be an option to leave that ApiController out of this PR and spend some time developing a administrative API together? 
Don't get me wrong, the idea of providing an administrative API is pretty good and the implementation might have been sufficient for covering needs during the hackathon, but for a more sophisticated API, which can be used by e.g. operators to promote resilience in production we should perhaps spend some more time for an appropriate design.
E.g. I somehow feel those endpoints should at least be exposed on another port to be shielded from the connectors main remote interfaces. Furthermore we need to find fitting authenticity checks of the calling user-agent etc.

Reply:
Lets move the `ApiController` code to the `samples/other` folder, so that we have it available for demos and PoCs etc.

Defining a production-ready public-facing REST API is no trivial task, we need to take care of authentication, authorization, possibly even rate-limiting (so as not to DDoS a connector), all that good stuff. I think for that another issue is in order :)